simulation/simtypes/account.go

```diff
@@ -39,9 +39,17 @@ func (sim *SimCtx) RandomSimAccountWithConstraint(f SimAccountConstraint) (simul
 	return sim.randomSimAccount(filteredAddrs), true
 }
 
-func (sim *SimCtx) RandomSimAccountWithMinCoins(ctx sdk.Context, denom string, amount int64) (simulation.Account, error) {
+func (sim *SimCtx) RandomSimAccountWithMinCoins(ctx sdk.Context, coins sdk.Coins) (simulation.Account, error) {
 	accHasMinCoins := func(acc simulation.Account) bool {
-		return sim.BankKeeper().GetBalance(ctx, acc.Address, denom).IsGTE(sdk.NewCoin(denom, sdk.NewInt(amount)))
+		spendableCoins := sim.BankKeeper().SpendableCoins(ctx, acc.Address)
+		for _, minCoin := range coins {
+			for _, spendableCoin := range spendableCoins {
+				if minCoin.Denom == spendableCoin.Denom {
+					return spendableCoin.IsGTE(sdk.NewCoin(minCoin.Denom, minCoin.Amount))
```

Comment:
This looks wrong, you'll return on the first matching coin, even if the rest are too small.

I think you want is return `spendableCoins.IsAllGTE(minCoins) && minCoins.DenomsSubsetOf(spendableCoins)`

Reply:
You are right, implemented suggestion, thank you! 🙏